src/connecpy/shared_client.py

```diff
@@ -66,3 +67,73 @@ def prepare_get_params(request_data, headers):
     if "content-encoding" in headers:
         params["compression"] = headers.pop("content-encoding")
     return params
+
+
+_current_response = ContextVar["Response"]("connecpy_current_response")
+
+
+def handle_response_headers(headers: Headers):
+    response = _current_response.get(None)
+    if not response:
+        return
+
+    response_headers: list[tuple[str, str]] = []
+    response_trailers: list[tuple[str, str]] = []
+    for key, value in headers.multi_items():
+        if key.lower().startswith("trailer-"):
+            key = key[len("trailer-") :]
+            obj = response_trailers
+        else:
+            obj = response_headers
+        obj.append((key, value))
+    if response_headers:
+        response._headers = Headers(response_headers)
+    if response_trailers:
+        response._trailers = Headers(response_trailers)
+
+
+class Response:
```

Comment:
The name `Response` might be confusing (at least to me) since it's not the actual response but a metadata collector. Perhaps `ResponseMetadata` or `ResponseMetadataCapture` or `ResponseContext` would be clearer?

```python
with ResponseMetadata() as metadata:
    resp = client.MakeHat(Size(inches=10))
    print(metadata.headers())
```

Reply:
Thanks - I was also unsure of the name but thought maybe we'll expose something other than metadata with it in the future. But probably not, so it seems like a better name indeed